client/test/e2e/shared/components.js

```diff
@@ -134,6 +134,15 @@ exports.currencyInput = {
   },
 
   /**
+  *set the value of the component by his id to avoid conflict
+  */
+  setById : function setById (id, value){
```

Comment:
You will  still be looking for the element by id.  It's just to reduce the component's test API and the amount of code we have to maintain.

I wrote a [wiki article](https://github.com/IMA-WorldHealth/bhima-2.X/wiki/Testing:-Creating-Component-Tests) to try and clarify this idea.  Let me know what you think.